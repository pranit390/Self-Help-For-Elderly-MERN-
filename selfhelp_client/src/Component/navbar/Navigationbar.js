import React from 'react';
import { Link,NavLink } from 'react-router-dom';
import { Nav, Navbar,Button } from 'react-bootstrap';
import styled from 'styled-components';
import  logo from '../image/mylogo.png'
import Swal from 'sweetalert2';

const Styles = styled.div`
.navbar { background-color: #C9FCDF;}
  a, .navbar-nav,.navdropdown .navbar-light .nav-link
  {
    font-size: 1.1em;
    font-style: normal;
    font-family: arial;
    font-weight:bold;
    border:none;
    &:hover {color: #103D60;}
  }
  .navbar-brand {
    font-size: 2.3em;
    font-weight:900;
    font-family:arial;
    &:hover {
    color:#103D60 }
    h5{
        font-weight:bold;
        font-style:normal;
        font-family:arial;
        font-size:1.1rem;
        text-align:center;
        text-decoration:none;
   }
  }
  .NavItems{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .inactive{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .active{
    color: black;
   }
`;

export const NavigationBar = () => (
  
  <Styles>
    <Navbar expand="lg">
    <Navbar.Brand>
                <img
                        src={logo}
                        width="60"
                        height="60"
                        className="d-inline align-top"
                    />
                    SELF HELP<br/><h5 className="heading">A Helping Hand</h5></Navbar.Brand>
      
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
         
            
              <NavLink to="/CustomerDashboard" className="inactive mx-2" activeClassName="active" exact={true}>Home</NavLink>
              <NavLink to="/CustomerDashboardTask" className="inactive mx-2" activeClassName="active" exact={true}>Tasks</NavLink>
              <NavLink to="/CustomerProfile" className="inactive mx-2" activeClassName="active" exact={true}>My Profile</NavLink>
              <NavLink to="/feedback" className="inactive mx-2" activeClassName="active" exact={true}>Rate Us</NavLink>
         
          
        
        </Nav>
      </Navbar.Collapse>

      <Button variant="danger" onClick={() => {
                    localStorage.removeItem("authToken");

                    Swal.fire({
                        title: 'Are you sure?',
                        text: 'Do you want to logout from the application!',
                        icon: 'warning',
                        showCancelButton: true,
                        confirmButtonText: 'Yes',
                        cancelButtonText: 'No'
                    }).then((result) => {
                        if (result.value) {
                            Swal.fire(
                                'Logged out!',
                            ) .then(function() {
                                localStorage.removeItem("adminInfo");
                                window.location.href = "/"
                            });
                        } else if (result.dismiss === Swal.DismissReason.cancel) {
                            Swal.fire(
                                'Cancelled',
                            )
                        }
                    })
                    
                    
                    }}>LOGOUT</Button>
    </Navbar>
  </Styles >
)