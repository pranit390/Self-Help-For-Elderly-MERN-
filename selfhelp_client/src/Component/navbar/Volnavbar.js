import React from 'react';
import axios from "axios";
import { Navbar, Nav, Container, Button, NavDropdown, Glyphicon } from 'react-bootstrap';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import  logo from '../image/mylogo.png';
import {NavLink,Link} from 'react-router-dom';
import Swal from 'sweetalert2';
import API_URL from "../../ApiHelper/API_URL.js";

const Styles = styled.div`

.navbar { background-color: #C9FCDF;}
  a, .navbar-nav,.navdropdown .navbar-light .nav-link
  {
    font-size: 1.1em;
    font-style: normal;
    font-family: arial;
    font-weight:bold;
    border:none;
    &:hover {color: #103D60;}
  }
  .navbar-brand {
    font-size: 2.3em;
    font-weight:900;
    font-family:arial;
    &:hover {
    color:#103D60 }
    h5{
        font-weight:bold;
        font-style:normal;
        font-family:arial;
        font-size:1.1rem;
        text-align:center;
        text-decoration:none;
   }
  }
  .NavItems{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .inactive{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .active{
    color: black;
   }

`;

 const Volnavbar = () => {

    return (
        <Styles>
            <Navbar fixed="top" expand="xl">
                <Navbar.Brand>
                <img
                        src={logo}
                        width="60"
                        height="60"
                        className="d-inline align-top"
                    />
                    SELF HELP<br/><h5 className="heading">A HELPING HAND</h5></Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto ml-4">
                        <NavLink to="/voldashboard" className="inactive mx-2" activeClassName="active" exact={true}>Home</NavLink>
                        <NavLink to="/volunteertask" className="inactive mx-2" activeClassName="active" exact={true}>Tasks</NavLink>
                        {/* <NavLink to="/rating" className="inactive mx-2" activeClassName="active" exact={true}>Rating</NavLink> */}
           
                        <NavLink to="/VolunteerProfile" className="inactive mx-2" activeClassName="active" exact={true}>My Profile</NavLink>
          
                       
              



                    </Nav>
                   
                </Navbar.Collapse>

                <Button variant="danger" onClick={() => {
                    

                    Swal.fire({
                        title: 'Are you sure?',
                        text: 'Do you want to logout from the application!',
                        icon: 'warning',
                        showCancelButton: true,
                        confirmButtonText: 'Yes',
                        cancelButtonText: 'No'
                    }).then((result) => {
                        if (result.value) {
                            Swal.fire(
                                'Logged out!',
                            ) .then(function() {

                                const fetchPrivateDate = async () => {
                                    const config = {
                                      headers: {
                                        "Content-Type": "application/json",
                                        Authorization: `Bearer ${localStorage.getItem("authToken")}`,
                                      },
                                    };
                              
                                    try {
                                      const  res  = await axios.get(`${API_URL}/volunteerAuth/getParticularVolunteer`, config);
                                     
                                      const payload = {
                                        availability:false
                                      }
                                      await axios.put(`${API_URL}/volunteer/updateVolunteerEngage/${res.data._id}`, payload)
                                     
                                    } catch (error) {
                                      localStorage.removeItem("authToken");
                                     // setError("You are not authorized please login");
                                    }
                                  };
                                 
                                  fetchPrivateDate();

                               setTimeout(()=>{localStorage.removeItem("authToken");
                               window.location.href = "/"},3000); 
                                
                            });
                        } else if (result.dismiss === Swal.DismissReason.cancel) {
                            Swal.fire(
                                'Cancelled',
                            )
                        }
                    })
                    
                    
                    }}>LOGOUT</Button>


            </Navbar>
        </Styles>
    )
}

export default Volnavbar;
