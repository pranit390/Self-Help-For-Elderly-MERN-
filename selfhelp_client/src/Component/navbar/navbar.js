import React from 'react'
import { Navbar, Nav,  NavDropdown } from 'react-bootstrap'
import styled from 'styled-components'
import 'bootstrap/dist/css/bootstrap.min.css';
import  logo from '../image/mylogo.png'
import {NavLink,Link} from 'react-router-dom'

const Styles = styled.div`

.navbar { background-color: #C9FCDF;}
  a, .navbar-nav,.navdropdown .navbar-light .nav-link
  {
    font-size: 1.1em;
    font-style: normal;
    font-family: arial;
    font-weight:bold;
    border:none;
    &:hover {color: #103D60;}
  }
  .navbar-brand {
    font-size: 2.3em;
    font-weight:900;
    font-family:arial;
    &:hover {
    color:#103D60 }
    h5{
        font-weight:bold;
        font-style:normal;
        font-family:arial;
        font-size:1.1rem;
        text-align:center;
        text-decoration:none;
   }
  }
  .NavItems{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .inactive{
    font-weight:bold;
    font-style:normal;
    font-family:arial;
    font-size:1.1rem;
    text-align:center;
    text-decoration:none;
    color:green;
   }
   .active{
    color: black;
   }

`;

const NavigationBar = () => {

    return (
        <Styles>
            <Navbar fixed="top" expand="xl">
                <Navbar.Brand>
                <img
                        src={logo}
                        width="60"
                        height="60"
                        className="d-inline align-top"
                    />
                    SELF HELP<br/><h5 className="heading">A HELPING HAND</h5></Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto ml-4">
                        <NavLink to="/" className="inactive mx-2" activeClassName="active" exact={true}>Home</NavLink>
                        <NavLink to="/AboutUs" className="inactive mx-2" activeClassName="active" exact={true}>About Us</NavLink>
                        <NavLink to="/ServiceList" className="inactive mx-2" activeClassName="active" exact={true}>Our Services</NavLink>
                    </Nav>
                    <Nav className="ml-auto">
                    <NavDropdown title="LOGIN" id="basic-nav-dropdown" className="NavItems">
                            <NavDropdown.Item><Link to="/Login" className="NavItems">Customer</Link></NavDropdown.Item>
                            <NavDropdown.Item><Link to="/VolunteerLogin" className="NavItems">Volunteer</Link></NavDropdown.Item>
                            <NavDropdown.Item><Link to="/Admin_Login" className="NavItems">Admin</Link></NavDropdown.Item>
                        </NavDropdown>
                        <NavDropdown title="SIGNUP" id="basic-nav-dropdown" className="NavItems">
                            <NavDropdown.Item><Link to="/customerRegistration" className="NavItems">Customer</Link></NavDropdown.Item>
                            <NavDropdown.Item><Link to="/volunteerRegistration" className="NavItems">Volunteer</Link></NavDropdown.Item>
                        </NavDropdown>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        </Styles>
    );
};

export default NavigationBar;
