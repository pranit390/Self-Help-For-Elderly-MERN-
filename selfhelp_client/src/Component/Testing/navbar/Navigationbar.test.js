import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import {shallow} from 'enzyme'

import {NavigationBar} from '../../navbar/Navigationbar';

describe("register customer", () => {
    it("should render my component", () => {
      const wrapper = shallow(<NavigationBar/>);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
});