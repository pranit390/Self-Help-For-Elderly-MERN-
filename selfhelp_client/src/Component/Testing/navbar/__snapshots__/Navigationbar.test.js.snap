// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register customer should render my component 1`] = `
Array [
  <styled.div>
    <Navbar
      collapseOnSelect={false}
      expand="lg"
      variant="light"
    >
      <NavbarBrand>
        <img
          className="d-inline align-top"
          height="60"
          src="mylogo.png"
          width="60"
        />
        SELF HELP
        <br />
        <h5
          className="heading"
        >
          A Helping Hand
        </h5>
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      />
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <Nav
          className="mr-auto"
          fill={false}
          justify={false}
        >
          <NavLink
            activeClassName="active"
            className="inactive mx-2"
            exact={true}
            to="/CustomerDashboard"
          >
            Home
          </NavLink>
          <NavLink
            activeClassName="active"
            className="inactive mx-2"
            exact={true}
            to="/CustomerDashboardTask"
          >
            Tasks
          </NavLink>
          <NavLink
            activeClassName="active"
            className="inactive mx-2"
            exact={true}
            to="/CustomerProfile"
          >
            My Profile
          </NavLink>
          <NavLink
            activeClassName="active"
            className="inactive mx-2"
            exact={true}
            to="/feedback"
          >
            Rate Us
          </NavLink>
        </Nav>
      </NavbarCollapse>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        variant="danger"
      >
        LOGOUT
      </Button>
    </Navbar>
  </styled.div>,
]
`;
